lib/internal/loader/Loader.js

```diff
@@ -63,31 +59,27 @@ class Loader {
       throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'url', 'string');
     }
 
-    let urlObj;
     if (format === 'builtin') {
-      urlObj = new URL(`node:${url}`);
-    } else {
-      urlObj = new URL(url);
-      if (urlObj.protocol !== 'file:' && urlObj.protocol !== 'node:') {
-        throw new errors.Error('ERR_INVALID_PROTOCOL',
-                               urlObj.protocol, 'file:');
-      }
+      return { url: `node:${url}`, format };
+    }
+
+    if (!url.startsWith('file:') && !url.startsWith('node:')) {
+      throw new errors.Error('ERR_INVALID_PROTOCOL', url, 'file:');
```

Comment:
Should there be a separate error for "node:" protocol or is that something internal only that users won't be able to use?

Reply:
i'm not sure / want userland experimentation here. `node:` seems fine, but I don't like occupying a protocol so kinda want it to be semi-discouraged. However, people still need to be able to make their resolver point to the built in modules. I am 50-50 on if removing the `node:` prefix is preferable to keeping it.

I don't want people to use `import 'node:util'` in their ESM right now though. So it is inconsistent which makes me :( , but don't have a better approach.

Reply:
I don't see the `node:` protocol being used for any kind of detection. It's parsed out and ignored as far as I can tell. Why not just use `file:` as with the others now that there is a "addon" hook.

Reply:
Ahh this is for builtin not addon modules. The `node:` prefix made me think it was for `.node` stuff.

Reply:
> However, people still need to be able to make their resolver point to the built in modules

What's the current way Node handles builtin module references as urls?

Reply:
builtin specifiers resolves to an internal `node:$name`, not exposed and does not allow query strings, etc. to builtin modules currently

Reply:
You mean [here](https://github.com/nodejs/node/blob/6c38ab34a301f76a2d5f306ddc242d2bebbd8f53/lib/internal/loader/resolveRequestUrl.js#L58-L78)?

If that's the only place then `node:` can be replace with `file:` since it's not being used *(it's ignored when parsing to just get the pathname)*

Reply:
@jdalton well spotted in that this path is no longer hit as it only applies to `builtin` case. Have removed the node check here. The `node:` scheme is not exposed to users of the resolver as the builtin scenario takes `{ format: 'builtin', url: 'fs' }` with `node:` scheme only used internally for these cases.